src/cardano/CardanoTools.ts

```diff
@@ -0,0 +1,191 @@
+import * as Cardano from '@emurgo/cardano-serialization-lib-nodejs'
+import { div } from 'biggystring'
+import { entropyToMnemonic, mnemonicToEntropy, validateMnemonic } from 'bip39'
+import {
+  EdgeCurrencyInfo,
+  EdgeCurrencyTools,
+  EdgeEncodeUri,
+  EdgeIo,
+  EdgeMetaToken,
+  EdgeParsedUri,
+  EdgeWalletInfo
+} from 'edge-core-js/types'
+import { base16 } from 'rfc4648'
+
+import { PluginEnvironment } from '../common/innerPlugin'
+import { encodeUriCommon, parseUriCommon } from '../common/uriHelpers'
+import { getLegacyDenomination } from '../common/utils'
+import {
+  asCardanoPrivateKeys,
+  asSafeCardanoWalletInfo,
+  CardanoNetworkInfo,
+  EpochParams,
+  SafeCardanoWalletInfo
+} from './cardanoTypes'
+
+export class CardanoTools implements EdgeCurrencyTools {
+  io: EdgeIo
+  currencyInfo: EdgeCurrencyInfo
+  networkInfo: CardanoNetworkInfo
+  epochParams?: EpochParams
+
+  constructor(env: PluginEnvironment<CardanoNetworkInfo>) {
+    const { currencyInfo, io, networkInfo } = env
+    this.currencyInfo = currencyInfo
+    this.io = io
+    this.networkInfo = networkInfo
+  }
+
+  derivePrivateKeys(mnemonic: string): {
+    accountKey: Cardano.Bip32PrivateKey
+    privateKey: Cardano.Bip32PrivateKey
+  } {
+    const mnemonicEntropy = mnemonicToEntropy(mnemonic)
+    const privateKey = Cardano.Bip32PrivateKey.from_bip39_entropy(
+      base16.parse(mnemonicEntropy),
+      new Uint8Array()
+    )
+
+    function harden(num: number): number {
+      return 0x80000000 + num
```

Comment:
I would do `0x80000000 | num` instead of `+`. It doesn't matter in this particular case, but it's just more correct.

Reply:
We can skip this one, since we want to match what the sample code is doing.